api/v1alpha1/guestosdescriptor_types.go

```diff
@@ -0,0 +1,58 @@
+// Copyright (c) 2020 VMware, Inc. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+//go:generate go run ./internal/generate_guestosidentifiers.go
+
+package v1alpha1
+
+// VirtualMachineGuestOSDescriptor is used to specify information about the
+//
+type VirtualMachineGuestOSDescriptor struct {
+	// VirtualMachineGuestOSIdentifier is the guest OS identifier.
+	//
+	// If ommited the Family field is used to determine a default
+	// VirtualMachineGuestOSIdentifier value.
+	//
+	// +optional
+	ID VirtualMachineGuestOSIdentifier `json:"id,omitempty"`
+
+	// VirtualMachineGuestOSFamily specifies the family of the guest OS.
+	//
+	// If omitted the VirtualMachineImage is used to attempt to determine
+	// the guest's family and/or ID.
+	//
+	// +optional
+	Family VirtualMachineGuestOSFamily `json:"family,omitempty"`
+}
+
+// VirtualMachineGuestOSFamily specifies the family of the guest OS.
+type VirtualMachineGuestOSFamily string
+
+const (
+	VirtualMachineGuestOSFamilyWindows       VirtualMachineGuestOSFamily = "windowsGuest"
+	VirtualMachineGuestOSFamilyLinux                                     = "linuxGuest"
```

Comment:
Do we also need to specify `VirtualMachineGuestOSFamily` type for all these consts, or is this Golang magic that automatically assigns them the types? Either way, I would vote for being explicit in the name of readability. (Same for Guest OS Identifiers)

Reply:
:-D 
I actually tried this in playground by checking the type of two consts, before leaving the comment.  I still thought you knew something that I didn't about Golang internals (as you often do).